code/client/client.cpp

```diff
@@ -101,6 +103,45 @@ std::vector<glm::vec3> colourSquare(std::vector<glm::vec3> dest, glm::vec3 colou
 
 }
 
+
+struct ExampleComponent
+{
+	int a{ 0 };
+	int b{ 0 };
+	int c{ 0 };
+};
+
+
+struct ExampleSystem : ecs::ISystem
+{
+
+	virtual void Initialize() {}
+	virtual void Teardown() {}
+	virtual void Update(ecs::Scene &scene, float deltaTime)
+	{
+		for (Guid entityGuid : ecs::EntitiesInScene<ExampleComponent>(scene))
+		{
+			// pass in the guid of the entity to lookup the corresponding component
+			ExampleComponent& ex = scene.GetComponent<ExampleComponent>(entityGuid);
+
+			// do some operation of the data
+			ex.a += 1;
+			ex.b += 2;
+			ex.c += 3;
+
+			display = "ecs component: " + std::to_string(ex.a) + " " + std::to_string(ex.b) + " " + std::to_string(ex.c) + "!";
+		}
+	}
+
+	std::string getDisplayString()
```

Comment:
pointless getter?

Reply:
perhaps simply a `to_string` method or someth would work, that way you're not storing a string in the component and you can just construct one on demand (actually this could be a fair perf improvement for a large number of these but idk i haven't been profiling stuff lol lets talk abt it).

Reply:
that code is just to illustrate API, so it'll probably be deleted pretty soon // performance isn't important.

Reply:
another example can be found here: https://github.com/dylanleclair/tonic-ecs/blob/main/client/client.cpp

Reply:
fair enough, I missed the forest for the trees. My bad